util/timer.h

```diff
@@ -124,6 +125,7 @@ class Timer {
       if (!running_) {
         return false;
       }
+      shutting_down_ = true;
       running_ = false;
```

Comment:
If we have two concurrent calls to Shutdown, one will return before the timer is fully shutdown, with tasks potentially happening. User might reasonably expect timer to be inactive when Shutdown returns (regardless of return value, assuming no additional Start).

Reply:
I agree.